### PR TITLE
Improvements to STAC implementation

### PR DESCRIPTION
**Pull request by pont-us**
This PR improves STAC support, providing full implementation of the `core`, `features`, and `collections` conformance classes, validated with [stac-api-validator](https://github.com/stac-utils/stac-api-validator). Additional conformance classes and/or extensions (in particular, the datacube extension) may be implemented in future PRs.

Addresses #818 .

Checklist:
* [x] Add unit tests and/or doctests in docstrings
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [x] New/modified features documented in `docs/source/*`
* [ ] ~Changes documented in `CHANGES.md`~ (initial implementation already documented there, nothing to add)
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/server/api.py [hunk 1]
```diff
@@ -593,11 +593,13 @@ def set_header(self, name: str, value: str):
...
     @abstractmethod
-    def write(self, data: Union[str, bytes, JSON]):
+    def write(self, data: Union[str, bytes, JSON],
+              content_type: Optional[str] = None):
         """Write data."""
 
     @abstractmethod
-    def finish(self, data: Union[str, bytes, JSON] = None):
+    def finish(self, data: Union[str, bytes, JSON] = None,
+               content_type: Optional[str] = None):
```
Comment (forman) [inline xcube/server/api.py:602]:
```suggestion
    def finish(self, 
               data: Union[str, bytes, JSON] = None,
               content_type: Optional[str] = None):
```

File: xcube/server/api.py [hunk 2]
```diff
@@ -593,11 +593,13 @@ def set_header(self, name: str, value: str):
         """Set the HTTP header *name* to given *value*."""
 
     @abstractmethod
-    def write(self, data: Union[str, bytes, JSON]):
+    def write(self, data: Union[str, bytes, JSON],
+              content_type: Optional[str] = None):
```
Comment (forman) [inline xcube/server/api.py:597]:
```suggestion
    def write(self, 
              data: Union[str, bytes, JSON],
              content_type: Optional[str] = None):
```

File: xcube/server/webservers/tornado.py [hunk 3]
```diff
@@ -480,9 +480,20 @@ def set_header(self, name: str, value: str):
     def set_status(self, status_code: int, reason: Optional[str] = None):
         self._handler.set_status(status_code, reason=reason)
 
-    def write(self, data: Union[str, bytes, JSON]):
-        self._handler.write(data)
-
-    def finish(self, data: Union[str, bytes, JSON] = None):
-        return self._handler.finish(data)
+    def write(self, data: Union[str, bytes, JSON],
+              content_type: Optional[str] = None):
```
Comment (forman) [inline xcube/server/webservers/tornado.py:484]:
```suggestion
    def write(self, 
              data: Union[str, bytes, JSON],
              content_type: Optional[str] = None):
```

File: xcube/server/webservers/tornado.py [hunk 4]
```diff
@@ -480,9 +480,20 @@ def set_header(self, name: str, value: str):
...
+        # https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.write
+        # "If the given chunk is a dictionary, we write it as JSON and set the
+        # Content-Type of the response to be application/json. (if you want to
+        # send JSON as a different Content-Type, call set_header after calling
+        # write())."
+        if content_type is not None:
+            self._handler.set_header('Content-Type', content_type)
+
+    def finish(self, data: Union[str, bytes, JSON] = None,
+               content_type: Optional[str] = None):
```
Comment (forman) [inline xcube/server/webservers/tornado.py:496]:
```suggestion
    def finish(self, 
               data: Union[str, bytes, JSON] = None,
               content_type: Optional[str] = None):
```

File: xcube/webapi/meta/routes.py [hunk 5]
```diff
@@ -66,9 +66,10 @@ class OpenApiJsonHandler(ApiHandler):
     )
     def get(self):
         include_all = self.request.get_query_arg('all', default=False)
-        self.response.finish(self.ctx.get_open_api_doc(
-            include_all=include_all
-        ))
+        self.response.finish(
+            self.ctx.get_open_api_doc(include_all=include_all),
+            content_type='application/vnd.oai.openapi+json;version=3.0'
```
Comment (forman) [inline xcube/webapi/meta/routes.py:71]: Nice!

**Review by forman — CHANGES_REQUESTED** [review]
Very good! 
Just a few but strong formatting suggestions.

**Comment by pont-us** [comment]
Thanks for the review, @forman . All suggestions accepted, will merge as soon as CI has re-run.

**Pull request merged**